src/Xsc/XscClient.ts

```diff
@@ -0,0 +1,60 @@
+import { HttpClient } from '../HttpClient';
+import { ILogger } from '../../model/';
+import { XscLogger } from './XscLogger';
+import { IClientSpecificConfig } from '../../model/ClientSpecificConfig';
+
+import { XscEventClient } from './XscEventClient';
+import { XscSystemClient } from './XscSystemClient';
+import { XrayScanClient } from '../Xray/XrayScanClient';
+
+export class XscClient {
+    private readonly httpClient: HttpClient;
+    private logger: ILogger;
+
+    public constructor(config: IClientSpecificConfig) {
+        const {
+            serverUrl,
+            logger = console,
+            username,
+            password,
+            accessToken,
+            proxy,
+            headers,
+            retries,
+            timeout,
+            retryOnStatusCode,
+            retryDelay,
+        }: IClientSpecificConfig = config;
+        if (!serverUrl) {
```

Comment:
shouldn't the validation of username+password OR accessToken should also be here?

Reply:
no, the only requirement is the url, validation is on the application side that defines the configurations